lib/src/inline_parser.dart

```diff
@@ -89,17 +91,26 @@ class InlineParser {
     while (!isDone) {
       var matched = false;
 
-      // See if any of the current tags on the stack match. We don't allow tags
-      // of the same kind to nest, so this takes priority over other possible
-      // matches.
-      for (var i = _stack.length - 1; i > 0; i--) {
-        if (_stack[i].tryMatch(this)) {
-          matched = true;
-          break;
+      var ch = charAt(pos);
+      if (ch == $rbracket ||
+          ch == $asterisk ||
+          ch == $underscore ||
+          ch == $tilde) {
+        // TODO(srawlins): This is imperfect as it doesn't take into account
+        // other possible syntax extensions. We can provide an API in the
+        // future where each syntax's endPattern can be used here.
+
+        // See if any of the current tags on the stack match.  This takes
+        // priority over other possible matches.
+        for (var i = _stack.length - 1; i > 0; i--) {
```

Comment:
We're iterating over all the elements except a position 0 though (a guard element).

I've update the PR so that this line isn't changed, so it's not part of this PR.